ui/app/src/context/UtcTimeZone.tsx

```diff
@@ -0,0 +1,41 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { createContext, useContext, useState } from 'react';
+import { TimeZoneProvider } from '@perses-dev/components';
+
+interface UtcTimeZoneContext {
+  isUtcTimeZone: boolean;
+  setUtcTimeZone: (value: boolean) => void;
+}
+
+export const UtcTimeZoneContext = createContext<UtcTimeZoneContext | undefined>(undefined);
+
+export function UtcTimeZoneProvider(props: { children: React.ReactNode }) {
+  const [isUtcTimeZone, setUtcTimeZone] = useState(false);
+  return (
+    <TimeZoneProvider timeZone={isUtcTimeZone ? 'utc' : 'local'}>
+      <UtcTimeZoneContext.Provider value={{ isUtcTimeZone, setUtcTimeZone }}>
+        {props.children}
+      </UtcTimeZoneContext.Provider>
+    </TimeZoneProvider>
```

Comment:
This threw me for a loop a little bit at first when reading it (with there being two providers). Did you consider doing this as one provider? Feels like maybe you didn't want to expose set functionality to all the consumers?

Reply:
You will notice this closely mirrors the app/src/context/DarkMode implementation.

The demonstration App can be implemented in various ways, supporting the ability to show and modify the timezone in the app header, as long as that timezone is made available via TimeZoneProvider to the embedded Perses dashboards. I thought it best to keep the Perses TimeZoneProvider minimal (read-only timezone state), and to implement the App support for view/edit timezone similar to its theme switcher for consistency.